src/pages/settings/Profile/PersonalDetails/LegalNamePage.tsx

```diff
@@ -1,82 +1,70 @@
 import * as ErrorUtils from '@libs/ErrorUtils';
 import Navigation from '@libs/Navigation/Navigation';
 import * as ValidationUtils from '@libs/ValidationUtils';
 import * as PersonalDetails from '@userActions/PersonalDetails';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PrivatePersonalDetails} from '@src/types/onyx';
+import type {Errors} from '@src/types/onyx/OnyxCommon';
 
-const propTypes = {
-    /* Onyx Props */
-
+type LegalNamePageOnyxProps = {
     /** User's private personal details */
-    privatePersonalDetails: PropTypes.shape({
-        legalFirstName: PropTypes.string,
-        legalLastName: PropTypes.string,
-    }),
-
-    ...withLocalizePropTypes,
+    privatePersonalDetails: OnyxEntry<PrivatePersonalDetails>;
 };
 
-const defaultProps = {
-    privatePersonalDetails: {
-        legalFirstName: '',
-        legalLastName: '',
-    },
-};
+type LegalNamePageProps = LegalNamePageOnyxProps;
 
-const updateLegalName = (values) => {
-    PersonalDetails.updateLegalName(values.legalFirstName.trim(), values.legalLastName.trim());
+const updateLegalName = (values: PrivatePersonalDetails) => {
+    PersonalDetails.updateLegalName(values.legalFirstName?.trim() ?? '', values.legalLastName?.trim() ?? '');
 };
 
-function LegalNamePage(props) {
+function LegalNamePage({privatePersonalDetails = {legalFirstName: '', legalLastName: ''}}: LegalNamePageProps) {
     const styles = useThemeStyles();
+    const {translate} = useLocalize();
     usePrivatePersonalDetails();
-    const legalFirstName = lodashGet(props.privatePersonalDetails, 'legalFirstName', '');
-    const legalLastName = lodashGet(props.privatePersonalDetails, 'legalLastName', '');
-    const isLoadingPersonalDetails = lodashGet(props.privatePersonalDetails, 'isLoading', true);
+    const legalFirstName = privatePersonalDetails?.legalFirstName ?? '';
+    const legalLastName = privatePersonalDetails?.legalLastName ?? '';
+    const isLoadingPersonalDetails = privatePersonalDetails?.isLoading ?? true;
 
-    const validate = useCallback((values) => {
-        const errors = {};
+    const validate = useCallback((values: PrivatePersonalDetails) => {
+        const errors: Errors = {};
 
-        if (!ValidationUtils.isValidLegalName(values.legalFirstName)) {
+        if (!ValidationUtils.isValidLegalName(values.legalFirstName ?? '')) {
             ErrorUtils.addErrorMessage(errors, 'legalFirstName', 'privatePersonalDetails.error.hasInvalidCharacter');
-        } else if (_.isEmpty(values.legalFirstName)) {
+        } else if (values.legalFirstName === '') {
             errors.legalFirstName = 'common.error.fieldRequired';
         }
-        if (ValidationUtils.doesContainReservedWord(values.legalFirstName, CONST.DISPLAY_NAME.RESERVED_NAMES)) {
+        if (ValidationUtils.doesContainReservedWord(values.legalFirstName ?? '', CONST.DISPLAY_NAME.RESERVED_NAMES as unknown as string[])) {
```

Comment:
I've checked and this function is used only with this value `CONST.DISPLAY_NAME.RESERVED_NAMES`, so maybe we can update the typing or just use by default, not sure...
```suggestion
        if (ValidationUtils.doesContainReservedWord(values.legalFirstName ?? '', CONST.DISPLAY_NAME.RESERVED_NAMES)) {
```